internal/db/schema/migrations/oss/postgres/59/01_target_ingress_egress_worker_filters.up.sql

```diff
@@ -0,0 +1,137 @@
+begin;
+
+-- Update tables to add ingress_worker_filter and egress_worker_filter
+alter table session
+  add column egress_worker_filter wt_bexprfilter,
+  add column ingress_worker_filter wt_bexprfilter;
+
+alter table target_tcp
+  add column egress_worker_filter wt_bexprfilter,
+  add column ingress_worker_filter wt_bexprfilter;
+
+-- Trigger functions to ensure that worker_filter and ingress/egress_worker_filter are mutually exclusive
+-- and that worker_filter can only be updated
+create function validate_filter_values_on_insert() returns trigger
+as $$
+begin
+  if new.worker_filter is not null then
+    raise exception 'worker_filter is deprecated and cannot be set';
+  end if;
+
+  return new;
+
+end;
+$$ language plpgsql;
+
+create function validate_filter_values_on_update() returns trigger
+as $$
+begin
+  if new.egress_worker_filter is not null then
+    if new.worker_filter = old.worker_filter then
+      new.worker_filter = null;
```

Comment:
We went back and forth on it but decided to clear the old filter instead of making them clear it manually, some of that discussion is in the thread here: https://hashicorp.slack.com/archives/C038L63F67L/p1667242927405219